src/ApiException.php

```diff
@@ -114,6 +114,39 @@ public static function createFromApiResponse(
         );
     }
 
+    /**
+     * @param string $basicMessage
+     * @param int $rpcCode
+     * @param array|null $metadata
+     * @param \Exception $previous
+     * @return ApiException
+     */
+    public static function createFromApiResponseREST(
+        $basicMessage,
+        $rpcCode,
+        array $metadata = null,
+        \Exception $previous = null
+    ) {
+        if (empty($metadata)) {
+            $decodedMetadata = [];
+            return self::create(
+                $basicMessage,
+                $rpcCode,
+                $metadata,
+                $decodedMetadata,
+                $previous
+            );
+        } else {
+            return self::create(
+                $basicMessage,
+                $rpcCode,
+                $metadata,
+                $metadata,
+                $previous
+            );
+        }
```

Comment:
implicit else can be used here because we return above
```suggestion
        }

        return self::create(
            $basicMessage,
            $rpcCode,
            $metadata,
            $metadata,
            $previous
        );
```